src/Products/SQLAlchemyDA/da.py

```diff
@@ -392,7 +392,7 @@ def query(self, query_string, max_rows=None, query_data=None):
                      (time.time() - ts_start))
 
         if desc is None:
-            return [], None
+            return [], []
```

Comment:
```suggestion
            return (), ()
```

Return tuples instead of lists, this is how it's done in ZMySQLDA[1]

[1] https://github.com/zopefoundation/Products.ZMySQLDA/blob/20bf2897181e06fa70b2d45cfebe4103bf7eee86/src/Products/ZMySQLDA/db.py#L595